internal/install/application_configuration.go

```diff
@@ -165,20 +167,41 @@ func selectElasticAgentImageName(version string) string {
 	if ok && strings.ToLower(valueEnv) != "true" {
 		disableWolfiImages = false
 	}
-	if !disableWolfiImages && !v.LessThan(elasticAgentWolfiVersion) {
+	switch {
+	case !disableWolfiImages && !v.LessThan(elasticAgentWolfiVersion) && agentBaseImage != "complete":
 		return elasticAgentWolfiImageName
-	}
-	if !v.LessThan(elasticAgentCompleteOwnNamespaceVersion) {
+	case !v.LessThan(elasticAgentCompleteOwnNamespaceVersion):
 		return elasticAgentCompleteImageName
-	}
-	if !v.LessThan(elasticAgentCompleteFirstSupportedVersion) {
+	case !v.LessThan(elasticAgentCompleteFirstSupportedVersion):
 		return elasticAgentCompleteLegacyImageName
+	default:
+		return elasticAgentImageName
+	}
+}
+
+type configurationOptions struct {
+	agentBaseImage string
+	stackVersion   string
+}
+
+type ConfigurationOption func(*configurationOptions)
+
+// OptionWithAgentBaseImage sets the agent image type to be used.
+func OptionWithAgentBaseImage(agentBaseImage string) ConfigurationOption {
+	return func(opts *configurationOptions) {
+		opts.agentBaseImage = agentBaseImage
+	}
+}
+
+// OptionWithStackVersion sets the Elastic Stack version to be used.
+func OptionWithStackVersion(stackVersion string) ConfigurationOption {
+	return func(opts *configurationOptions) {
+		opts.stackVersion = stackVersion
 	}
-	return elasticAgentImageName
 }
 
 // Configuration function returns the elastic-package configuration.
-func Configuration() (*ApplicationConfiguration, error) {
+func Configuration(options ...ConfigurationOption) (*ApplicationConfiguration, error) {
```

Comment:
Does it work without a stack version? Should it be mandatory?
```suggestion
func Configuration(stackVersion string, options ...ConfigurationOption) (*ApplicationConfiguration, error) {
```

Reply:
While doing these changes, there was an usage where it was not used stack version:
https://github.com/elastic/elastic-package/blob/d8572d9f9a0b378cecc67c9789a16bfbdf37173a/cmd/profiles.go#L81

In that command, it looks like that it just checked the config file in the `~/.elastic-package` directory. It's not needed there the stack versions to retrieve the images of the services.